src/T2DMap.cpp

```diff
@@ -2324,7 +2342,7 @@ void T2DMap::paintMapInfo(const QElapsedTimer& renderTimer, QPainter& painter, c
         testRect = painter.boundingRect(
                 mMapInfoRect.left() + 10, mMapInfoRect.top() + 10, mMapInfoRect.width() - 20, mMapInfoRect.height() - 20, Qt::TextWordWrap | Qt::AlignLeft | Qt::AlignTop, infoText);
 
-    } while ((testRect.height() > mMapInfoRect.height() - 20 || testRect.width() > mMapInfoRect.width() - 20) && infoHeight < height());
+    } while ((testRect.height() > mMapInfoRect.height() - mFontHeight || testRect.width() > mMapInfoRect.width() - 20) && infoHeight < height());
```

Comment:
The `20` is from the `10` margin on all four sides - note the comment: `// Test in a rectangle that is 10 less on all sides:` as such it should not, I think,  be changed to `mFontHeight`...!

The purpose of this `do` loop is to size the shaded rectangle so that the provided text fits within it - I haven't determined yet whether you account for this in your override - have you tried making the map widget narrow enough that the text will not fit in in 3 (or 4 in development version that also show the rendering time)?

Reply:
And yet there is huge gap at the bottom which I've tried to get rid of... Mhm I had an illusion that it made things better, but they are actually identical. No better... no worse :) Will revert that one and take attempt at that another time.